RateLimiter/ApiRule/Factory/RateLimitedFactory.cs

```diff
@@ -0,0 +1,37 @@
+﻿using RateLimiter.ApiRule.Factory;
+using RateLimiter.ApiRule.RuleImplementations;
+using RateLimiter.Model.Enum;
+using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace RateLimiter.ApiRule.Factory
+{
+    public class RateLimitedFactory : IRateLimitedFactory
+    {
+        public List<IRuleValidation> GetRateLimitRulesByResource(ResourceEnum resourceName)
```

Comment:
The purpose is returning rule's list of  objects acrording to the resource name. I agree with you that external code can add/remove items in returned list of rules.
So it would be better to use IEnumerable instead of List.